uportal-war/src/main/java/org/jasig/portal/url/UrlSyntaxProviderImpl.java

```diff
@@ -1091,7 +1099,11 @@ protected void addPortletUrlData(
             final String portletWindowIdStr = portletWindowId.toString();
             prefixedPortletWindowId = SEPARATOR + portletWindowIdStr;
             suffixedPortletWindowId = portletWindowIdStr + SEPARATOR;
-            url.addParameter(PARAM_ADDITIONAL_PORTLET, portletWindowIdStr);
+
+            // handle URI (i.e. search) that requires skipping additional portlet info
+            if (!request.getRequestURI().contains(this.compactParamsForUriFragment)) {
+                url.addParameter(PARAM_ADDITIONAL_PORTLET, portletWindowIdStr);
+            }
```

Comment:
The original approach failed three unit tests in UrlSyntaxProviderImplTest. The tests checked for the additional parameters. I am reviewing those for commonality. All three explicitly set "action=dashboard".

So simply checking if portletUrlBuilder.getCopyCurrentRenderParameters() returned a value was insufficient. I will add a check for "action=dashboard" and test.